de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/NegotiationHandler.java

```diff
@@ -0,0 +1,80 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import org.apache.log4j.Logger;
+import de.fu_berlin.inf.dpp.negotiation.SessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.OutgoingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.ProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.AbstractOutgoingProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.AbstractIncomingProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.NegotiationTools;
+import de.fu_berlin.inf.dpp.server.progress.ConsoleProgressIndicator;
+import de.fu_berlin.inf.dpp.session.INegotiationHandler;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+
+public class NegotiationHandler implements INegotiationHandler {
+
+    private static final Logger LOG = Logger
+        .getLogger(NegotiationHandler.class);
+
+    private final ISarosSessionManager sessionManager;
+
+    public NegotiationHandler(ISarosSessionManager sessionManager) {
+        sessionManager.setNegotiationHandler(this);
+        this.sessionManager = sessionManager;
+    }
+
+    @Override
+    public void handleOutgoingSessionNegotiation(
+        OutgoingSessionNegotiation negotiation) {
+        SessionNegotiation.Status status = negotiation
+            .start(new ConsoleProgressIndicator(System.out));
+
+        switch (status) {
+        case OK:
+            sessionManager.startSharingProjects(negotiation.getPeer());
+            break;
+        case CANCEL:
+        case REMOTE_CANCEL:
+            break;
+        case ERROR:
+        case REMOTE_ERROR:
+            LOG.error(negotiation.getErrorMessage());
+            break;
+        }
+    }
+
+    @Override
+    public void handleIncomingSessionNegotiation(
+        IncomingSessionNegotiation negotiation) {
+        negotiation.localCancel("Server does not accept session invites",
+            NegotiationTools.CancelOption.NOTIFY_PEER);
+    }
+
+    @Override
+    public void handleOutgoingProjectNegotiation(
+        AbstractOutgoingProjectNegotiation negotiation) {
+        ProjectNegotiation.Status status = negotiation
+            .run(new ConsoleProgressIndicator(System.out));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=144&issues=AWSovkpxMDLKieEST687&open=AWSovkpxMDLKieEST687)

[ISSUE_KEY]: AWSovkpxMDLKieEST687